https_client.cpp

```diff
@@ -1,14 +1,27 @@
-#include <SDKDDKVer.h>
+#ifdef _WIN32
+# include <SDKDDKVer.h>
+#endif
 
 #include <boost/asio.hpp>
-#include <boost/asio/windows_sspi.hpp>
+
+#ifdef _WIN32
+# include <boost/asio/windows_sspi.hpp>
+#else
+# include <boost/asio/ssl.hpp>
+#endif
 
 #include <cstdlib>
 #include <iostream>
 #include <iterator>
 #include <string>
 #include <fstream>
 
+#ifdef _WIN32
+namespace ssl = boost::asio::windows_sspi;
+#else
+namespace ssl = boost::asio::ssl;
+#endif
+
```

Comment:
Consider using
```
namespace net = boost::asio;
```

Reply:
Worth considering. I'll also consider removing `using boost::asio::ip::tcp;` since I personally hate pulling in symbols without a qualified name, but these things can most likely be considered related.